### PR TITLE
Initial Setup of bmplib

### PR DESCRIPTION
**Pull request by dschaffner**
Added setup.py, setup.cfg, pyproject.toml, and two __init__.py files for bmplib and spectral subpackage

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+long_description_content_type = text/markdown
+license = 3-clause BSD
+license_file = LICENSES/LICENSE.txt
+url = https://github.com/BaPSF/bapsflib
+download_url = https://pypi.org/project/bapsflib/
+project_urls =
+    BaPSF = http://plasma.physics.ucla.edu/
+    Documentation = https://bapsflib.readthedocs.io/en/latest/
+    GitHub = https://github.com/BaPSF/bapsflib
+keywords = bapsf, HDF5, lapd, physics, plasma, plasma physics, science
```
Comment (cacsphysics) [inline setup.cfg]: I am unsure if I can edit, but all instances of bapsf should be changed to bmplib.

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+python_requires = >=3.6
+zip_safe = false
+packages = find:
+include_package_data = True
+setup_requires =
+    # PEP-518 encourages the use of pyproject.toml for defining
+    # build dependencies...they should be defined under build-system.requires
+install_requires =
+    # ought to mirror requirements/install.txt
+    astropy >= 3.1
```
Comment (cacsphysics) [inline setup.cfg]: If we are going to start incorporating plasmapy, then the install should require plasmapy.

File: bmplib/__init__.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            f"bmplib.__version__ not generated (set to 'unknown'), bmplib is "
+            f"not an installed package and {warn_add}.",
+            RuntimeWarning,
+        )
+
+        del warn
+    del fallback_version, warn_add
+
+
+del pkg_resources
```
Comment (rocco8773) [inline bmplib/__init__.py:55]:
```suggestion
del pkg_resources

```

It's PEP8 standard to end all `.py` files with a blank line.

File: bmplib/spectral/__init__.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+#
+#
+#
+
+"""
+The spectral analysis subpackage
+"""
+__all__ = ["spectrum_wwind"]
+
+from bmplib.spectral import spectrum_wwind
```
Comment (rocco8773) [inline bmplib/spectral/__init__.py:14]:
```suggestion
from bmplib.spectral import spectrum_wwind

```

File: setup.cfg (hunk 1 above)
Comment (cacsphysics) [reply setup.cfg]: Ignore this I see that you changed them

File: setup.cfg (hunk 2 above)
Comment (cacsphysics) [reply setup.cfg]: I see that you changed them.

File: bmplib/spectral/spectrum_wwind.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
+#compute FFT of given array
+
+import numpy as np
+from scipy.signal import blackman,bartlett,hanning,hamming
```
Comment (rocco8773) [inline bmplib/spectral/spectrum_wwind.py:4]:
```suggestion
import numpy as np

from scipy.signal import blackman,bartlett,hanning,hamming
```

It's good practice to split direct import and from-import into separate groups.

File: bmplib/spectral/spectrum_wwind.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+def spectrum_wwind(array,time,window='None'):#time should be in seconds
```
Comment (rocco8773) [inline bmplib/spectral/spectrum_wwind.py:6]:
```suggestion
def spectrum_wwind(array, time, window='None'):  # time should be in seconds
```

PEP8 standards

File: bmplib/spectral/spectrum_wwind.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    #Size of array
+    Nw = array.shape[0]
+    
+    #Calculate time step (assumed to be in seconds)
+    dt = time[1]-time[0]
```
Comment (rocco8773) [inline bmplib/spectral/spectrum_wwind.py:11]:
```suggestion
    dt = time[1] - time[0]
```

PEP8

File: bmplib/spectral/spectrum_wwind.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    
+    #prefactor
+    #print 'dt = ',dt
+    prefactor = dt
+    
+    #Calculate array of frequencies, shift
+    w = np.fft.fftfreq(Nw,dt)
```
Comment (rocco8773) [inline bmplib/spectral/spectrum_wwind.py:18]:
```suggestion
    w = np.fft.fftfreq(Nw, dt)
```

File: bmplib/spectral/spectrum_wwind.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    
+    #Calcuate Phase
+    phase = np.angle(aw)
+    phase0 = np.fft.fftshift(phase)
+    
+    #Adjust arrays if not div by 2
+    if not np.mod(Nw,2):
+        w0 = np.append(w0,-w0[0])
+        aw0 = np.append(aw0,-aw0[0])
+        phase0 = np.append(phase0,-phase0[0])
```
Comment (rocco8773) [inline bmplib/spectral/spectrum_wwind.py:47]:
```suggestion
    if not np.mod(Nw, 2):
        w0 = np.append(w0, -w0[0])
        aw0 = np.append(aw0, -aw0[0])
        phase0 = np.append(phase0, -phase0[0])
```

File: bmplib/spectral/spectrum_wwind.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    
+    comp = aw
+    pwr = (np.abs(aw2))**2
+    pwr2 = (np.abs(aw))**2
+    mag = np.sqrt(pwr)
+    cos_phase = np.cos(phase2)
+    freq = w2
+    freq2 = w
+    
+    return freq,freq2,comp,pwr,mag,phase2,cos_phase,dt
```
Comment (rocco8773) [inline bmplib/spectral/spectrum_wwind.py:63]:
```suggestion
    return freq, freq2, comp, pwr, mag, phase2, cos_phase, dt

```

PEP8

File: pyproject.toml.py [hunk 11]
```diff
@@ -0,0 +1,7 @@
+[build-system]
+requires = [
+    "setuptools >= 41.2",
+    "setuptools_scm",
+    "wheel >= 0.29.0",
+]  # ought to mirror 'requirements/build.txt'
+build-backend = "setuptools.build_meta"
```
Comment (rocco8773) [inline pyproject.toml.py:7]:
```suggestion
build-backend = "setuptools.build_meta"

```

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+zip_safe = false
+packages = find:
+include_package_data = True
+setup_requires =
+    # PEP-518 encourages the use of pyproject.toml for defining
+    # build dependencies...they should be defined under build-system.requires
+install_requires =
+    # ought to mirror requirements/install.txt
+    #astropy >= 3.1
+    h5py >= 2.6
```
Comment (rocco8773) [inline setup.cfg:48]: I'm guessing this a preemptive dependency.

File: setup.cfg [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    #%(extras)s
+    #sphinx <= 2.4.4
+    #sphinx-automodapi
+    #sphinx-gallery
+    #sphinx_rtd_theme
+developer =
+    # install everything for developers
+    #%(docs)s
+    #%(extras)s
+    #%(tests)s
```
Comment (rocco8773) [inline setup.cfg:74]:
```suggestion
    #%(tests)s

```

File: setup.py [hunk 14]
```diff
@@ -0,0 +1,18 @@
+
+#!/usr/bin/env python3
```
Comment (rocco8773) [inline setup.py:2]:
```suggestion
#!/usr/bin/env python3
```

File: setup.py [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+#
+import os
+
+from setuptools import setup
+
+# find here
+here = os.path.abspath(os.path.dirname(__file__))
+
+# ---- Perform setup                                                        ----
+setup(use_scm_version=True)
```
Comment (rocco8773) [inline setup.py:18]:
```suggestion
setup(use_scm_version=True)

```

**Review by rocco8773 — COMMENTED** [review]
It looks good to me.  I made a few PEP8 formatting suggestions.  It takes awhile to learn them all so I thought I'd just point them out.

I don't think it's necessary for this PR, but you might want to consider adding the `requiremtns.txt` files.

**Pull request merged**